test/download-dist

```diff
@@ -48,11 +48,17 @@ def download_dist(wait=False):
         message("download-dist: pre-built dist/ are for NODE_ENV=production")
         return False
 
+    # If the HEAD commit is a merge (when checking out a pull request /merge ref), use the merged
+    # commit ID to find the cached data, instead of the merge itself
     try:
-        sha = subprocess.check_output(["git", "rev-parse", "HEAD"], stderr=subprocess.DEVNULL).decode().strip()
+        sha = subprocess.check_output(["git", "rev-parse", "--verify", "HEAD^2"], stderr=subprocess.DEVNULL).decode().strip()
```

Comment:
I'd be nervous about this, without some kind of assurance that the thing we download was built for (the equivalent of) the tree that we're actually inside of.  This specifically works in the cockpit case because we do a rebase to get our tree into the exact same shape as the one that built `dist/`.  Here I guess we don't do that.

Reply:
Specifically: the merge (which represents the state of the current tree) very possibly has a lot of commits in it that weren't present when `dist/` was built.